client/client.go

```diff
@@ -343,16 +346,125 @@ func configureAwsClient(ctx context.Context, logger hclog.Logger, awsConfig *Con
+	ListAccounts(ctx context.Context, params *organizations.ListAccountsInput, optFns ...func(*organizations.Options)) (*organizations.ListAccountsOutput, error)
+}
+
+func getOUAccounts(ctx context.Context, accountsApi ListAccountsAPI, ous []string) ([]orgTypes.Account, error) {
+	var rawAccounts []orgTypes.Account
+
+	for _, ou := range ous {
+		var paginationToken *string
+		for {
+			resp, err := accountsApi.ListAccountsForParent(ctx, &organizations.ListAccountsForParentInput{
+				NextToken: paginationToken,
+				ParentId:  aws.String(ou),
+			})
+			if err != nil {
+				return nil, err
+			}
+			rawAccounts = append(rawAccounts, resp.Accounts...)
+			if resp.NextToken == nil {
+				break
+			}
+			paginationToken = resp.NextToken
+		}
+	}
+
+	return rawAccounts, nil
+}
+
+func getAllAccounts(ctx context.Context, accountsApi ListAccountsAPI) ([]orgTypes.Account, error) {
+	var rawAccounts []orgTypes.Account
+	var paginationToken *string
+
+	for {
+		resp, err := accountsApi.ListAccounts(ctx, &organizations.ListAccountsInput{
+			NextToken: paginationToken,
+		})
+		if err != nil {
+			return nil, err
+		}
+		rawAccounts = append(rawAccounts, resp.Accounts...)
+		if resp.NextToken == nil {
+			break
+		}
+		paginationToken = resp.NextToken
+	}
+	return rawAccounts, nil
+
+}
+
+func loadAccounts(ctx context.Context, accountsApi ListAccountsAPI, ous []string) ([]orgTypes.Account, error) {
+	if len(ous) > 0 {
+		return getOUAccounts(ctx, accountsApi, ous)
+	}
+	return getAllAccounts(ctx, accountsApi)
+
+}
+
+func loadOrgAccounts(ctx context.Context, logger hclog.Logger, awsConfig *Config) ([]Account, AssumeRoleAPIClient, error) {
+
+	adminAccount := Account{AccountName: "Default-Admin-Account"}
+	if awsConfig.Organization.AdminAccount != nil {
+		adminAccount = *awsConfig.Organization.AdminAccount
+	}
+	awsCfg, err := configureAwsClient(ctx, logger, awsConfig, adminAccount, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+	accounts := make([]Account, 0)
+	svc := organizations.NewFromConfig(awsCfg)
+	rawAccounts, err := loadAccounts(ctx, svc, awsConfig.Organization.OrganizationUnits)
+	if err != nil {
+		return nil, nil, err
+	}
+	for _, account := range rawAccounts {
+		logger.Info("adding account", "account", account)
+		// Only load Active accounts
+		if account.Status != orgTypes.AccountStatusActive {
+			continue
+		}
+		accounts = append(accounts, Account{
+			ID:              *account.Id,
+			RoleARN:         fmt.Sprintf("arn:aws:iam::%s:role/%s", *account.Id, awsConfig.Organization.ChildAccountRoleName),
+			RoleSessionName: awsConfig.Organization.ChildAccountRoleSessionName,
+			ExternalID:      awsConfig.Organization.ChildAccountExternalID,
+			LocalProfile:    awsConfig.Organization.AdminAccount.LocalProfile,
+			Regions:         awsConfig.Organization.ChildAccountRegions,
+			source:          "org",
+		})
+	}
+	return accounts, sts.NewFromConfig(awsCfg), err
+
+}
+
+func (org AwsOrg) Validate() bool {
```

Comment:
`IsValid`? If not valid, should we explicitly return an error? So maybe return an `error` to pass on the message? Not sure

Reply:
I just decided to remove the entire function